pkgs/by-name/te/technitium-dns-server/package.nix

```diff
@@ -29,7 +50,7 @@ buildDotnetModule rec {
   # move dependencies from TechnitiumLibrary to the expected directory
   preBuild = ''
     mkdir -p ../TechnitiumLibrary/bin
-    cp -r ${technitium-dns-server-library}/lib/${technitium-dns-server-library.pname}/* ../TechnitiumLibrary/bin/
+    cp -r ${technitium-library}/lib/TechnitiumLibrary/* ../TechnitiumLibrary/bin/
```

Comment:
```suggestion
    cp -r ${technitium-library}/lib/TechnitiumLibrary/* ../${technitium-library.pname}/bin/
```

Reply:
Why do you think this might be helpful? I assume that the library will not be renamed by the original author?!

Reply:
the path will be whatever you have as pname for the package in the let binding, this will not change if you keep them in sync (you probably wouldn't change this ever, realisticly) but in the spirit of keeping it DRY I would reference that here instead of writing it again. the target path in the server program needs the library path to match whatever the builddotnetmodule sets it to and that will set it to the pname. so this does not depend on what the original author named it. you can try this out by changing the pname but not the reference here.

Reply:
OK, I think I got what you mean. However, I just created that directory with the line above it. 
```
mkdir -p ../TechnitiumLibrary/bin
```

So in case we want to replace it, we'd need to replace it in both lines. Correct? I see your argument. However, it makes the code a little less easy to read:

```
  preBuild = ''
    mkdir -p ../${technitium-library.pname}/bin
    cp -r ${technitium-library}/lib/TechnitiumLibrary/* ../${technitium-library.pname}/bin/
  '';
```

Reply:
Yes, correct. In the end it is up to you, I did that because when I extracted the library into its own package I had to dig down the layers of abstraction in builddotnetmodule (or more realistically: I opened a nix develop shell and looked around) because the build failed and that would be avoided by this change, hence my suggestion.

Reply:
I updated the PR accordingly -- in both places. :-)

Reply:
I am sorry for the confusion, I think the source dir needs to be set to the pname and you are absolutely right, the target dir which you create is whatever the upstream package name is. Sorry, I got confused by the diff..

Reply:
I replaced the wrong part in my suggestion, the source `TechnitiumLibrary` part should have been replaced by the library pname, sorry I'm on mobile at the moment and that males it really hard to parse the diffs